### PR TITLE
Use slf4j as logging facade (#2799)

### PR DESCRIPTION
**Pull request by devcsrj**
Also includes a maven enforcer plugin that halts the build when usage of java.util.logging.** are found.

In accordance to the [`SLF4JBridgeHandler`](https://www.slf4j.org/apidocs/org/slf4j/bridge/SLF4JBridgeHandler.html), I mapped the log levels into:
```
 FINEST  -> TRACE
 FINER   -> DEBUG
 FINE    -> DEBUG
 INFO    -> INFO
 WARNING -> WARN
 SEVERE  -> ERROR
```

As per the _concrete_ implementation, I used `log4j` as the default for test scope (defined in `flow-project` [pom.xml](https://github.com/vaadin/flow/pull/3049/files#diff-600376dffeb79835ede4a0b285078036R319), and `flow-documentation` [pom.xml](https://github.com/vaadin/flow/pull/3049/files#diff-4854a498ecb4b3e9a97b4c2e03b4b236R121).

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/flow/3049)

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/server/communication/PushHandler.java [hunk 1]
```diff
@@ -73,9 +73,9 @@
     private final PushEventCallback establishCallback = new PushEventCallback() {
         @Override
         public void run(AtmosphereResource resource, UI ui) throws IOException {
-            getLogger().log(Level.FINER,
-                    "New push connection for resource {0} with transport {1}",
-                    new Object[] { resource.uuid(), resource.transport() });
+            getLogger().debug(
+                    "New push connection for resource {} with transport {}",
+                    resource.uuid(), resource.transport());
```
[the remark this answers is not in the document]
Comment (devcsrj) [reply flow-server/src/main/java/com/vaadin/server/communication/PushHandler.java]: This *does* invoke the method conditionally. Actual humans, help?

File: flow-server/src/main/java/com/vaadin/ui/polymertemplate/DefaultTemplateParser.java [hunk 2]
```diff
@@ -64,8 +64,7 @@
             clazz -> new AtomicBoolean());
 
     @Override
-    public Element getTemplateContent(Class<? extends PolymerTemplate<?>> clazz,
-            String tag) {
+    public Element getTemplateContent(Class<? extends PolymerTemplate<?>> clazz, String tag) {
```
[the remark this answers is not in the document]
Comment (devcsrj) [reply flow-server/src/main/java/com/vaadin/ui/polymertemplate/DefaultTemplateParser.java]: Didn't change the behavior of this line whatsoever. The message is too vague too

File: flow-server/src/main/java/com/vaadin/server/communication/PushHandler.java [hunk 3]
```diff
@@ -114,7 +114,7 @@
      * respond to the request directly.)
      */
     private final PushEventCallback receiveCallback = (resource, ui) -> {
-        getLogger().log(Level.FINER, "Received message from resource {0}",
+        getLogger().debug("Received message from resource {}",
                 resource.uuid());
```
[the remark this answers is not in the document]
Comment (jreznot) [reply flow-server/src/main/java/com/vaadin/server/communication/PushHandler.java]: It is not required with SLF4J

File: flow-server/src/main/java/com/vaadin/server/communication/PushHandler.java [hunk 4]
```diff
@@ -70,10 +70,10 @@
...
      * now.
      */
-    private final PushEventCallback establishCallback = (resource, ui) -> {
-        getLogger().log(Level.FINER,
-                "New push connection for resource {0} with transport {1}",
-                new Object[] { resource.uuid(), resource.transport() });
+    private final PushEventCallback establishCallback = ( resource,  ui) -> {
+            getLogger().debug(
+                    "New push connection for resource {} with transport {}",
+                     resource.uuid(), resource.transport() );
```
[the remark this answers is not in the document]
Comment (jreznot) [reply flow-server/src/main/java/com/vaadin/server/communication/PushHandler.java]: It is not required with SLF4J

**Comment by jreznot** [comment]
It seems that Sonar is stupid about SLF4J Logger and does not know about conditional formatting

**Comment by SomeoneToIgnore** [comment]
Don't worry about it, it's mostly to help the reviewers.

---

Reviewed 28 of 51 files at r1, 5 of 10 files at r2, 19 of 20 files at r3.
Review status: all files reviewed at latest revision, 29 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/flow/3049#-L-kE9vo7hvjbxdGz0lA:-L-kE9vo7hvjbxdGz0lB:b-7kiap5) (no related file):*
Hey, thanks a lot for the contribution!

I've reviewed it and it looks almost fine, only small style adjustments are left.
There are two main groups of them:
* conditional formatting is not used everywhere 
* some `if` blocks lack curly braces around them

I've commented all the places I've spotted, don't bother to reply to them, it's mostly for your convenience, so that you can locate the code lines easily, just fix them if I am right. 
If not – then comment it and we'll discuss it.

---

*[flow-server/src/main/java/com/vaadin/router/InternalServerError.java, line 55 at r3](https://reviewable.io:443/reviews/vaadin/flow/3049#-L-kAe5ReJL6q5KQB7QF:-L-k5KY7Jqtd9JuLRRGY:b95td0t) ([raw file](https://github.com/vaadin/flow/blob/3eb99aaa9a15a5105b0267c8cb3bbf31f7e39a86/flow-server/src/main/java/com/vaadin/router/InternalServerError.java#L55)):*
> ```Java
>             String message = String.format( EXCEPTION_TRYING_NAVIGATE_S,
>                 event.getLocation().getPath() );
>             getLogger().error(message, exception);
> ```

Should we rather use conditional formatting here?

---

*[flow-server/src/main/java/com/vaadin/server/BootstrapHandler.java, line 861 at r3](https://reviewable.io:443/reviews/vaadin/flow/3049#-L-kAOyfPevruj8SGEEd:-L-kAOyfPevruj8SGEEe:be3wloa) ([raw file](https://github.com/vaadin/flow/blob/3eb99aaa9a15a5105b0267c8cb3bbf31f7e39a86/flow-server/src/main/java/com/vaadin/server/BootstrapHandler.java#L861)):*
> ```Java
>         writePushIdUIDL(json, session);
>         if (getLogger().isDebugEnabled()) {
>             getLogger().debug("Initial UIDL:" + json.asString());
> ```

I propose to use conditional formatting here either.

---

*[flow-server/src/main/java/com/vaadin/server/DefaultDeploymentConfiguration.java, line 442 at r3](https://reviewable.io:443/reviews/vaadin/flow/3049#-L-kAh7sjNBFn-NpfidS:-L-kAh7sjNBFn-NpfidT:bj21fqq) ([raw file](https://github.com/vaadin/flow/blob/3eb99aaa9a15a5105b0267c8cb3bbf31f7e39a86/flow-server/src/main/java/com/vaadin/server/DefaultDeploymentConfiguration.java#L442)):*
> ```Java
>                         + "'%s' value is not absolute (doesn't start with '/')",
>                     WEB_COMPONENTS_LOADER_JS_NAME, Constants.FRONTEND_URL_ES6 ) );
>             getLogger().warn(message);
> ```

This also looks like conditional formatting case, doesn't it?

---

*[flow-server/src/main/java/com/vaadin/server/DefaultDeploymentConfiguration.java, line 478 at r3](https://reviewable.io:443/reviews/vaadin/flow/3049#-L-kAoLALKHPamtuYs--:-L-kAoLALKHPamtuYs-0:b-2to1cy) ([raw file](https://github.com/vaadin/flow/blob/3eb99aaa9a15a5105b0267c8cb3bbf31f7e39a86/flow-server/src/main/java/com/vaadin/server/DefaultDeploymentConfiguration.java#L478)):*
> ```Java
>         String message = formatDefaultPolyfillMessage( String.format( "Will use %s polyfill discovered in %s",
>                 WEB_COMPONENTS_LOADER_JS_NAME, dirName ) );
>         getLogger().info(message);
> ```

Am I missing something, or this is also a nice candidate for conditional formatting?

---

*[flow-server/src/main/java/com/vaadin/server/VaadinService.java, line 1305 at r3](https://reviewable.io:443/reviews/vaadin/flow/3049#-L-kB34ykgdxw8m7e6Pv:-L-kB34ykgdxw8m7e6Pw:b8a2kqd) ([raw file](https://github.com/vaadin/flow/blob/3eb99aaa9a15a5105b0267c8cb3bbf31f7e39a86/flow-server/src/main/java/com/vaadin/server/VaadinService.java#L1305)):*
> ```Java
>                     getLogger().debug(
>                             "Closing inactive UI #{} in session {}",
>                             new Object[] { ui.getUIId(), sessionId });
> ```

Can we get rid of `new Object[]` here and use the method with two parameters instead?

---

*[flow-server/src/main/java/com/vaadin/server/communication/JSR356WebsocketInitializer.java, line 136 at r3](https://reviewable.io:443/reviews/vaadin/flow/3049#-L-kBcKg6uqZz1kDnLv9:-L-kBcKg6uqZz1kDnLvA:b-394s7z) ([raw file](https://github.com/vaadin/flow/blob/3eb99aaa9a15a5105b0267c8cb3bbf31f7e39a86/flow-server/src/main/java/com/vaadin/server/communication/JSR356WebsocketInitializer.java#L136)):*
> ```Java
>                     getLogger().warn(
>                             "Failed to initialize Atmosphere for "
>                                     + servletName,
> ```

Let's use conditional formatting here also.

---

*[flow-server/src/main/java/com/vaadin/server/communication/JSR356WebsocketInitializer.java, line 165 at r3](https://reviewable.io:443/reviews/vaadin/flow/3049#-L-kBggdpcD3RmnWfgM9:-L-kBggdpcD3RmnWfgMA:b-394s7z) ([raw file](https://github.com/vaadin/flow/blob/3eb99aaa9a15a5105b0267c8cb3bbf31f7e39a86/flow-server/src/main/java/com/vaadin/server/communication/JSR356WebsocketInitializer.java#L165)):*
> ```Java
>             return;
>         }
>         getLogger().debug("Creating AtmosphereFramework for " + servletName);
> ```

Let's use conditional formatting here also.

---

*[flow-server/src/main/java/com/vaadin/server/communication/JSR356WebsocketInitializer.java, line 169 at r3](https://reviewable.io:443/reviews/vaadin/flow/3049#-L-kBjg_7vpQwaCsI5qV:-L-kBjg_7vpQwaCsI5qW:b-394s7z) ([raw file](https://github.com/vaadin/flow/blob/3eb99aaa9a15a5105b0267c8cb3bbf31f7e39a86/flow-server/src/main/java/com/vaadin/server/communication/JSR356WebsocketInitializer.java#L169)):*
> ```Java
>                 new FakeServletConfig(servletRegistration, servletContext));
>         servletContext.setAttribute(attributeName, framework);
>         getLogger().debug("Created AtmosphereFramework for " + servletName);
> ```

Let's use conditional formatting here also.

---

*[flow-server/src/main/java/com/vaadin/server/communication/PushHandler.java, line 78 at r2](https://reviewable.io:443/reviews/vaadin/flow/3049#-Kzo8g9EDMeN5RkkVRbb-r2-78:-L-kBp8gnHCSt7twla4F:b-2cj7fa) ([raw file](https://github.com/vaadin/flow/blob/b23890e2d2d72f7c0f403f227439f0ec67fb3225/flow-server/src/main/java/com/vaadin/server/communication/PushHandler.java#L78)):*
<details><summary><i>Previously, devcsrj wrote…</i></summary><blockquote>

This *does* invoke the method conditionally. Actual humans, help?
</blockquote></details>

Actual humans understand that, worry not.

---

*[flow-server/src/main/java/com/vaadin/server/communication/PushHandler.java, line 76 at r3](https://reviewable.io:443/reviews/vaadin/flow/3049#-Kzo8g9EDMeN5RkkVRbb-r3-76:-L-kBy8jZhK6LnIby5uO:b-afrg6d) ([raw file](https://github.com/vaadin/flow/blob/3eb99aaa9a15a5105b0267c8cb3bbf31f7e39a86/flow-server/src/main/java/com/vaadin/server/communication/PushHandler.java#L76)):*
<details><summary><i>Previously, jreznot (Yuriy Artamonov) wrote…</i></summary><blockquote>

It is not required with SLF4J
</blockquote></details>

Funny that only this call is noted by Sonar.

Disregard the message, will be suppressed before the merge by me.

---

*[flow-server/src/main/java/com/vaadin/server/communication/PushHandler.java, line 373 at r3](https://reviewable.io:443/reviews/vaadin/flow/3049#-L-kCEfrHA2HXb6qkvaf:-L-kCEfs7MvKG5mzgyeH:b-zcfmui) ([raw file](https://github.com/vaadin/flow/blob/3eb99aaa9a15a5105b0267c8cb3bbf31f7e39a86/flow-server/src/main/java/com/vaadin/server/communication/PushHandler.java#L373)):*
> ```Java
>                 getLogger().warn(
>                         "Could not find push connection to close: {} with transport {}",
>                         new Object[] { id, resource.transport() });
> ```

Let's use conditional formatting here also, no need to create an array with slf4j.

---

*[flow-server/src/main/java/com/vaadin/server/communication/PushHandler.java, line 389 at r3](https://reviewable.io:443/reviews/vaadin/flow/3049#-L-kCHrJhYv7C6EaMcd_:-L-kCHrJhYv7C6EaMcda:b-394s7z) ([raw file](https://github.com/vaadin/flow/blob/3eb99aaa9a15a5105b0267c8cb3bbf31f7e39a86/flow-server/src/main/java/com/vaadin/server/communication/PushHandler.java#L389)):*
> ```Java
>                     getLogger().debug(
>                             "Connection unexpectedly closed for resource {} with transport {}",
>                             new Object[] { id, resource.transport() });
> ```

Let's use conditional formatting here also.

---

*[flow-server/src/main/java/com/vaadin/server/communication/PushRequestHandler.java, line 89 at r3](https://reviewable.io:443/reviews/vaadin/flow/3049#-L-kCO5YFzsk3TYuyD_K:-L-kCO5YFzsk3TYuyD_L:b-394s7z) ([raw file](https://github.com/vaadin/flow/blob/3eb99aaa9a15a5105b0267c8cb3bbf31f7e39a86/flow-server/src/main/java/com/vaadin/server/communication/PushRequestHandler.java#L89)):*
> ```Java
>         if (atmosphere == null) {
>             // Not initialized by JSR356WebsocketInitializer
>             getLogger().debug("Initializing Atmosphere for servlet "
> ```

Let's use conditional formatting here also.

---

*[flow-server/src/main/java/com/vaadin/server/communication/PushRequestHandler.java, line 101 at r3](https://reviewable.io:443/reviews/vaadin/flow/3049#-L-kCNaOP3SZnJUIoZbU:-L-kCNaOP3SZnJUIoZbV:b-394s7z) ([raw file](https://github.com/vaadin/flow/blob/3eb99aaa9a15a5105b0267c8cb3bbf31f7e39a86/flow-server/src/main/java/com/vaadin/server/communication/PushRequestHandler.java#L101)):*
> ```Java
>             }
>         } else {
>             getLogger().debug("Using pre-initialized Atmosphere for servlet "
> ```

Let's use conditional formatting here also.

---

*[flow-server/src/main/java/com/vaadin/server/communication/ServerRpcHandler.java, line 277 at r3](https://reviewable.io:443/reviews/vaadin/flow/3049#-L-kCRseMNbXoJppJO3w:-L-kCRseMNbXoJppJO3x:b-394s7z) ([raw file](https://github.com/vaadin/flow/blob/3eb99aaa9a15a5105b0267c8cb3bbf31f7e39a86/flow-server/src/main/java/com/vaadin/server/communication/ServerRpcHandler.java#L277)):*
> ```Java
>             } else {
>                 getLogger().warn(
>                         "Unexpected message id from the client. Expected: "
> ```

Let's use conditional formatting here also.

---

*[flow-server/src/main/java/com/vaadin/server/communication/StreamRequestHandler.java, line 150 at r3](https://reviewable.io:443/reviews/vaadin/flow/3049#-L-kCWtEJe6AyWR1e53H:-L-kCWtEJe6AyWR1e53I:b-394s7z) ([raw file](https://github.com/vaadin/flow/blob/3eb99aaa9a15a5105b0267c8cb3bbf31f7e39a86/flow-server/src/main/java/com/vaadin/server/communication/StreamRequestHandler.java#L150)):*
> ```Java
>         boolean hasPrefix = index >= 0;
>         if (!hasPrefix) {
>             getLogger().info("Unsupported path structure, path=" + path);
> ```

Let's use conditional formatting here also.

---

*[flow-server/src/main/java/com/vaadin/server/communication/StreamRequestHandler.java, line 165 at r3](https://reviewable.io:443/reviews/vaadin/flow/3049#-L-kCXdOIV79O4LzXRNi:-L-kCXdOIV79O4LzXRNj:b-394s7z) ([raw file](https://github.com/vaadin/flow/blob/3eb99aaa9a15a5105b0267c8cb3bbf31f7e39a86/flow-server/src/main/java/com/vaadin/server/communication/StreamRequestHandler.java#L165)):*
> ```Java
>             throw new RuntimeException(e);
>         } catch (URISyntaxException e) {
>             getLogger().info("Path '" + path
> ```

Let's use conditional formatting here also.

---

*[flow-server/src/main/java/com/vaadin/server/communication/rpc/AbstractRpcInvocationHandler.java, line 47 at r3](https://reviewable.io:443/reviews/vaadin/flow/3049#-L-kCf2-Un6do2zOmzYg:-L-kCf2-Un6do2zOmzYh:b-fk4pwv) ([raw file](https://github.com/vaadin/flow/blob/3eb99aaa9a15a5105b0267c8cb3bbf31f7e39a86/flow-server/src/main/java/com/vaadin/server/communication/rpc/AbstractRpcInvocationHandler.java#L47)):*
> ```Java
>         }
>         if (!node.isAttached()) {
>             getLogger().warn("Got an RPC for detached node: {}w",
> ```

`w` looks like an extra symbol that should be removed.

---

*[flow-server/src/main/java/com/vaadin/server/startup/RouteRegistry.java, line 428 at r3](https://reviewable.io:443/reviews/vaadin/flow/3049#-L-kClUKhWGSzFFDI7JW:-L-kClUKhWGSzFFDI7JX:b-394s7z) ([raw file](https://github.com/vaadin/flow/blob/3eb99aaa9a15a5105b0267c8cb3bbf31f7e39a86/flow-server/src/main/java/com/vaadin/server/startup/RouteRegistry.java#L428)):*
> ```Java
>                         "Registering route '%s' to navigation target '%s'.",
>                         alias, navigationTarget.getName());
>                 logger.debug(message);
> ```

Let's use conditional formatting here also.

---

*[flow-server/src/main/java/com/vaadin/ui/Html.java, line 151 at r3](https://reviewable.io:443/reviews/vaadin/flow/3049#-L-kCx-iauqlafh3zn2T:-L-kCx-iauqlafh3zn2U:b-fjfysw) ([raw file](https://github.com/vaadin/flow/blob/3eb99aaa9a15a5105b0267c8cb3bbf31f7e39a86/flow-server/src/main/java/com/vaadin/ui/Html.java#L151)):*
> ```Java
>     }
> 
>     private static Logger getLogger() {
> ```

Why do we need this in the class at all? It's not used, let's remove it instead.

---

*[flow-server/src/main/java/com/vaadin/ui/polymertemplate/DefaultTemplateParser.java, line 67 at r2](https://reviewable.io:443/reviews/vaadin/flow/3049#-Kzo8g9EDMeN5RkkVRbr-r2-67:-L-kDGZLZhuBvxOAI8FI:bxfs8kf) ([raw file](https://github.com/vaadin/flow/blob/b23890e2d2d72f7c0f403f227439f0ec67fb3225/flow-server/src/main/java/com/vaadin/ui/polymertemplate/DefaultTemplateParser.java#L67)):*
<details><summary><i>Previously, devcsrj wrote…</i></summary><blockquote>

Didn't change the behavior of this line whatsoever. The message is too vague too
</blockquote></details>

Should this belong to the other line? Anyway, the file changes look fine to me.

---

*[flow-server/src/main/java/com/vaadin/ui/polymertemplate/DefaultTemplateParser.java, line 100 at r3](https://reviewable.io:443/reviews/vaadin/flow/3049#-L-kDVNBRn8-XQGrOPeq:-L-kDVNC2p7FvCtqDCmS:b-9pkja4) ([raw file](https://github.com/vaadin/flow/blob/3eb99aaa9a15a5105b0267c8cb3bbf31f7e39a86/flow-server/src/main/java/com/vaadin/ui/polymertemplate/DefaultTemplateParser.java#L100)):*
> ```Java
>             String path = resolvePath(request, url);
> 
>             if (logEnabled)
> ```

Please use curly braces (`{}`) for the if block.

---

*[flow-server/src/main/java/com/vaadin/ui/polymertemplate/DefaultTemplateParser.java, line 109 at r3](https://reviewable.io:443/reviews/vaadin/flow/3049#-L-kDccEYuasuqVoJbPA:-L-kDccF-q_BLA_ggMCI:b-9pkja4) ([raw file](https://github.com/vaadin/flow/blob/3eb99aaa9a15a5105b0267c8cb3bbf31f7e39a86/flow-server/src/main/java/com/vaadin/ui/polymertemplate/DefaultTemplateParser.java#L109)):*
> ```Java
>                 }
>                 Element templateElement = parseHtmlImport(content, url, tag);
>                 if (logEnabled && templateElement != null)
> ```

Please use curly braces (`{}`) for the if block.

---

*[flow-server/src/main/java/com/vaadin/ui/polymertemplate/DefaultTemplateParser.java, line 119 at r3](https://reviewable.io:443/reviews/vaadin/flow/3049#-L-kDdrwuVBCftHL1g_M:-L-kDdrwuVBCftHL1g_N:b-9pkja4) ([raw file](https://github.com/vaadin/flow/blob/3eb99aaa9a15a5105b0267c8cb3bbf31f7e39a86/flow-server/src/main/java/com/vaadin/ui/polymertemplate/DefaultTemplateParser.java#L119)):*
> ```Java
>             } catch (IOException exception) {
>                 // ignore exception on close()
>                 if (logEnabled)
> ```

Please use curly braces (`{}`) for the if block.

---

*[flow-server/src/main/java/com/vaadin/util/ResponseWriter.java, line 98 at r3](https://reviewable.io:443/reviews/vaadin/flow/3049#-L-kDojQ0ikgqh4790Nm:-L-kDojQ0ikgqh4790Nn:b-gwlzbg) ([raw file](https://github.com/vaadin/flow/blob/3eb99aaa9a15a5105b0267c8cb3bbf31f7e39a86/flow-server/src/main/java/com/vaadin/util/ResponseWriter.java#L98)):*
> ```Java
>                 }
>             } catch (Exception e) {
>                 getLogger().debug(
> ```

Let's use conditional formatting here.

---

*[flow-spring-addon/pom.xml, line 88 at r3](https://reviewable.io:443/reviews/vaadin/flow/3049#-L-kDzWW5d9GHDlcNjxk:-L-kDzWW5d9GHDlcNjxl:b-73m6vj) ([raw file](https://github.com/vaadin/flow/blob/41d133d29a2310c1ea6600d6e4e79a47f7616503/flow-spring-addon/pom.xml#L88)):*
> ```XML
> 
>         <dependency>
>             <groupId>junit</groupId>
> ```

Nice catch.

---

*[flow-test-util/src/main/java/com/vaadin/flow/testutil/ViewOrUITest.java, line 103 at r3](https://reviewable.io:443/reviews/vaadin/flow/3049#-L-kE3nhpx9pJ8nU5OFk:-L-kE3niDEEv9w9ZDBDv:b-gwlzbg) ([raw file](https://github.com/vaadin/flow/blob/3eb99aaa9a15a5105b0267c8cb3bbf31f7e39a86/flow-test-util/src/main/java/com/vaadin/flow/testutil/ViewOrUITest.java#L103)):*
> ```Java
>             // Here only to please Sonar...
>             getLogger().error(
>                     " for " + getClass().getName() + " not found", e);
> ```

Let's use conditional formatting here.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3049#-:-L-kABmz31ePe8K4nCdd:b-wx75nu)*

**Comment by devcsrj** [comment]
Updated the PR!

(On a side note, it's a little sad Reviewable doesn't update the addressed lines as _changed_, like you'd normally see when making comments directly on GH.)

**Comment by SomeoneToIgnore** [comment]
Yeah, the tool is a bit weird, I agree.
What it does instead is builds supplementary commits to show the diffs after the last review, so it's kind of the same in most of the cases.

Anyway, awesome job, thanks a lot.
We will merge it tomorrow, right after the new release is made, since it's a breaking change and we don't want it to happen right before the release.

---

Reviewed 15 of 15 files at r4.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3049#-:-L-lDs2q7F498ui6Wd_F:br0b21l)*

**Comment by SomeoneToIgnore** [comment]
Ok, unused imports sound like a nice thing to be removed.
Otherwise don't worry about the Sonar.

---

Review status: all files reviewed at latest revision, 6 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3049#-:-L-lG28aS020Ef-YXQ66:bl58nqf)*

**Comment by SomeoneToIgnore** [comment]
Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3049)*

**Comment by SomeoneToIgnore** [comment]
Reviewed 1 of 2 files at r6.
Review status: 51 of 52 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3049)*

**Comment by SomeoneToIgnore** [comment]
Reviewed 1 of 2 files at r6.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3049)*

**Comment by SomeoneToIgnore** [comment]
Reviewed 2 of 2 files at r7.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3049)*

**Pull request merged**